netbox_topology_views/views.py

```diff
@@ -200,14 +201,29 @@ def create_node(
         if not group_id:
             print('Exception occured while handling default group.')
             return node
-   
+
     group = get_object_or_404(CoordinateGroup, pk=group_id)
 
-    node["physics"] = True
-    # Coords must be set even if no coords have been stored. Otherwise nodes with coords 
-    # will not be placed correctly by vis-network.
-    node["x"] = 0
-    node["y"] = 0
+    node["physics"] = not disable_physics
+    if not disable_physics:
+        # Coords must be set even if no coords have been stored. Otherwise nodes with coords
+        # will not be placed correctly by vis-network.
+        node["x"] = 0
+        node["y"] = 0
+    else:
+        # draw devices clustered by their rack neighbours
+        # for center coordinates of a rack, use the rack name as seed (if available)
+        random.seed(device.rack.name if hasattr(device, "rack") else device.name)
+        # set the upper size of the graph
+        plot_size = int(nnodes / 12) + 1
+        base_coords_x = random.randint(-plot_size * nnodes, plot_size * nnodes)
+        base_coords_y = random.randint(-plot_size * nnodes, plot_size * nnodes)
+        # make seed unique for devices for spread
+        if hasattr(device, "rack"):
+            random.seed(device.name)
+        # spread devices around the rack center coordinates
```

Comment:
Randomization will probably lead to a different layout when exporting to XML. Have you tested that?

Reply:
Yes, where previously all coordinates were (0, 0), they now are set to some values.
The diff between the export of "physics enabled" and "physics disabled" consists of all `mxGeometry` tags. When `save_coordinates` is enabled, the diff before and after moving a single node (and the same enabled/disabled setting for physics) is only one `mxGeometry` tag.

Re-rendering with disabled physics (hence with "random coordinates") yields the same file every time.